### DISCUSSION TITLE
What are the fields that uniquely identify an account transaction?

### DISCUSSION THREAD

**Discussion by stevebissett** [Investec-Developer-Community/.github#19] [Q&A]
**Question 1:** 
Is there a reason why a unique identifier is not provided by the API to uniquely identify an account transaction? (even a unique identifier within a given account)? 

**Question 2**
Given that an ID is not provided, which is the minimum set of fields that will always uniquely identify a transaction within an account so that a unique identifier can be built from this to help with an update or insert workflow. 

**Extra Context:**

There are some very similar questions and discussions (unanswered), perhaps with the same intent: 

Unique IDs are absent in several places throughout the API (account transactions, documents being two places).
I'm looking to understand which fields can be definitively used to uniquely identify a transaction. 

There is an example in the Ruby client here, which assumes amount + description + date can uniquely identify a transaction.

I know that some terminals prevent the same amount on the same day, but I don't think that this is always the case, and my assumption is that it's possible to have the same transaction description + amount + date on the same day. 

I assume that once these transactions are made, the `postedOrder` would indicate that they are indeed 2 different transactions. 

```ruby
    # At this point, there is no unique identifier being returned from Investec's API.
    # This method serves to create a stable unique identifier based on the transaction details.
    def id
      [
        amount.to_i,
        description,
        date.to_s
      ].map(&:to_s).join('-')
    end
```

I'm implementing a very common workflow: 
* Retrieve transactions periodically to store full history to include older transactions than the API allows. 
* Update transactions if they already exist and changes have been made to them.
* Create a new transaction if the transaction doesn't exist.
* Avoid duplicate transactions as an absolute requirement.

Related Discussions References

#10 
https://github.com/orgs/Investec-Developer-Community/discussions/10

#9
https://github.com/orgs/Investec-Developer-Community/discussions/9

#6 
https://github.com/orgs/Investec-Developer-Community/discussions/6

**Comment by nikspyratos** [comment]
I can say for Q 2, what I've been using is a combination of the account ID, date, description, currency, and amount. My use case was a bit simpler and I generally assume I won't be making the exact same transaction twice in the same day. If you add post order to that field list and to an upsert it should work.

**Reply by stevebissett** [reply]
I am currently doing this (using the postedOrder) - but I have a second ID that excludes it.
The main concern I have is that there are discussions around exposing pending transactions. 

I am unsure if pending transactions are still regarded as POSTED. 
I wouldn't want any fields that can change to be included in an identifier.

**Comment by programmable-banking-community** [comment]
[FOR VIS] Longer community discussion and context in our [Slack workspace here](https://investec-dev-com.slack.com/archives/C05MQQDUJ3E/p1718703607907239).
